torch/_inductor/codecache.py

```diff
@@ -1126,27 +1152,39 @@ def load(cls, source_code):
                     )
                     compile_file(input_path, output_path, cmd)
                 cls.cache[key] = cls._load_library(output_path)
-                cls.cache[key].key = key
+                cls.cache[key].key = key  # type: ignore[attr-defined]
 
         return cls.cache[key]
 
 
 class PyCodeCache:
-    cache: Dict[Any, types.ModuleType] = dict()
-    linemaps = dict()
+    cache: Dict[str, ModuleType] = dict()
+    linemaps: Dict[str, List[Tuple[Any, ...]]] = dict()
     clear = staticmethod(cache.clear)
 
     @classmethod
-    def write(cls, source_code, extra=""):
+    def write(cls, source_code: str, extra: str = "") -> Tuple[str, str]:
         return write(source_code, "py", extra=extra)
 
     @classmethod
-    def load(cls, source_code, extra="", linemap=()):
+    def load(
+        cls,
+        source_code: str,
+        extra: str = "",
+        linemap: Optional[List[Tuple[int, Optional[str]]]] = None,
+    ) -> ModuleType:
         key, path = write(source_code, "py", extra=extra)
         return cls.load_by_key_path(key, path, linemap)
 
     @classmethod
-    def load_by_key_path(cls, key, path, linemap=()):
+    def load_by_key_path(
+        cls,
+        key: str,
+        path: str,
+        linemap: Optional[List[Tuple[int, Optional[str]]]] = None,
```

Comment:
The default was previously an empty tuple, but the linemap is actually a list. That happened to work because the `zip` usage "did the right" thing and created an empty list when operating on an empty tuple. But I changed this to a real list and then obeyed the linter when it (correctly) doesn't want an empty list as a default param.